src/components/AnnotationOverlay/AnnotationOverlay.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { withTranslation } from 'react-i18next';
+
+import core from 'core';
+import selectors from 'selectors';
+
+import './AnnotationOverlay.scss';
+
+const MAX_WORDS = 100;
+
+const AnnotationOverlay = () => {
+  const isDisabled = useSelector(state => 
+    selectors.isElementDisabled(state, 'annotationOverlay')
+  );
+  const [isOpen, setIsOpen] = useState(false);
+  const [annotation, setAnnotation] = useState();
+  const [repliesCount, setRepliesCount] = useState(0);
+  const [overlayPosition, setOverlayPosition] = useState({
+    left: 0,
+    top: 0
+  });
+  
+  const am = core.getAnnotationManager();
+  
+  useEffect(() => {
+    window.docViewer.on('mouseMove.hover', onMouseHover);
+    return () => {
+      window.docViewer.off('mouseMove.hover', onMouseHover);
+    }
+  }, []);
+    
+  const onMouseHover = (e, mouseEvent) => {
+    const annotation = am.getAnnotationByMouseEvent(mouseEvent);
+    setIsOpen(!!annotation);
+    if (annotation) {
+      setAnnotation(annotation);
+      setOverlayPosition({
+        left: mouseEvent.clientX + 20,
+        top: mouseEvent.clientY + 20
+      });
+      setRepliesCount(annotation.getReplies().length);
```

Comment:
Can you move this function into the `effect` hook? If the this handler is using any props or state, being in the same function would be clearer to see what dependencies we need to pass to the `useEffect`